assets/js/blocks/product-collection/inspector-controls/attributes-control.tsx

```diff
@@ -0,0 +1,81 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useEffect, useState } from '@wordpress/element';
+import ProductAttributeTermControl from '@woocommerce/editor-components/product-attribute-term-control';
+import { AttributeMetadata } from '@woocommerce/types';
+import { SearchListItem } from '@woocommerce/editor-components/search-list-control/types';
+import {
+	ExternalLink,
+	// @ts-expect-error Using experimental features
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalToolsPanelItem as ToolsPanelItem,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { ProductCollectionQuery } from '../types';
+
+const EDIT_ATTRIBUTES_URL =
+	'/wp-admin/edit.php?post_type=product&page=product_attributes';
+
+interface Props {
+	woocommerceAttributes?: AttributeMetadata[];
+	setQueryAttribute: ( value: Partial< ProductCollectionQuery > ) => void;
+}
+
+const AttributesControl = ( {
+	woocommerceAttributes,
+	setQueryAttribute,
+}: Props ) => {
+	const [ selected, setSelected ] = useState< { id: number }[] >( [] );
+
+	useEffect( () => {
+		if ( woocommerceAttributes ) {
+			setSelected(
+				woocommerceAttributes.map( ( { termId: id } ) => ( {
+					id,
+				} ) )
+			);
+		}
+	}, [ woocommerceAttributes ] );
```

Comment:
Do you think we could convert this component to a fully controlled component so it doesn't need to hold its state? Something along these lines:

```suggestion
	const selected = woocommerceAttributes.map( ( { termId: id } ) => ( {
		id,
	} ) );
```

Reply:
That's a valid idea. By converting the component to a fully controlled component and using the mapped value of woocommerceAttributes for selected, we can eliminate the need for the useState hook. 🤝 

I made the changes in [f9a6813](https://github.com/woocommerce/woocommerce-blocks/pull/9600/commits/f9a6813300b2b9a3aa89f170f7339a4e4efaddc8).